Jenkinsfile

```diff
@@ -273,6 +275,7 @@ pipeline {
         Full_test       = " -DMIOPEN_TEST_ALL=On"
         Tensile_build_env = "MIOPEN_DEBUG_HIP_KERNELS=0 "
         Tensile_setup = " -DMIOPEN_TEST_MIOTENSILE=ON -DMIOPEN_USE_MIOPENTENSILE=ON -DMIOPEN_USE_ROCBLAS=OFF"
+        gfx90a_flags = "-DMIOPEN_TEST_GFX90a=ON"
```

Comment:
Remove this flag. Cmake should use auto detection

Reply:
Autodetection is only possible when #1099 is fully checked in.
Currently all CI nodes are using ROCm4.2, where rocminfo cannot recognize gfx90a properly thus causing GPU Detection to fail.
The "workaround" in this PR is to identify GPU type from arch ID, and pull ROCm4.3 only if gfx90a is set.
Sort of chicken and egg paradox: we need ROCm4.3 to detect gfx90a properly and then pull ROCm4.3 to build the docker :)
#1099 is going to fix the problem. However, since upgrading the ROCm version for all CI might be a big change, I hope to check in this one to unlock a lot of gfx90a related PRs. 
Thanks!

Reply:
but wait. This flag used only in 4 stages, but gfx90a was tested in 7 stages. So, the 3 stages work with autodetection and work fine.

Reply:
For this PR ROCm4.3 required only on gfx90a, docker rocm selection based on value in 'gpu_arch'. 
So no need to use (gfx90a_flags = "-DMIOPEN_TEST_GFX90a=ON") 
As I can see from the logs, the GPU was identified correctly.

Reply:
Yes, you are right :) gpu_arch and gfx90a_flags are redundant checks.
Let me make the changes and remove this flag. Thanks! @shurale-nkn